storage/storage.cpp

```diff
@@ -768,18 +777,32 @@ void Storage::OnServerListDownloaded(vector<string> const & urls)
   if (m_queue.empty())
     return;
 
-  QueuedCountry const & queuedCountry = m_queue.front();
-  TCountryId const & countryId = queuedCountry.GetCountryId();
-  MapOptions const file = queuedCountry.GetCurrentFile();
+  QueuedCountry & queuedCountry = m_queue.front();
+  if (queuedCountry.GetInitOptions() == MapOptions::Diff)
+  {
+    using diff_scheme::Status;
+    auto const status = m_diffManager.Status();
+    if (status == Status::Undefined)
+    {
+      m_diffManager.SetDeferredUpdateTask(std::async(std::launch::deferred,
```

Comment:
So, if m_diffManager.Status() is undefined, you defer call to the method. But why not use Observers? Looks like this is a hacky version, why not to rewrite this to clean Observers? I mean, you need to add Observer interface to the diff manager, and then subscribe here.